cmd/bootstrap/runner.go

```diff
@@ -562,3 +570,56 @@ func (r *runner) installOperator(ctx context.Context, helmClient helmclient.Inte
 
 	return nil
 }
+
+func (r *runner) patchChartOperatorDeployment(ctx context.Context, k8sClients k8sclient.Interface) error {
+	chartOperatorDeployment := "chart-operator-unique"
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waiting for deployment %#q", chartOperatorDeployment))
+
+	o := func() error {
+		deploy, err := k8sClients.K8sClient().AppsV1().Deployments(namespace).Get(ctx, chartOperatorDeployment, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return microerror.Maskf(executionFailedError, "deployment %#q in %#q not found", chartOperatorDeployment, namespace)
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
```

Comment:
Why not `return microerror.Mask(err)` in both? We would see `Deployment not found` in the error I guess? Not blocking because of this though.

Reply:
Sometimes we need to match if the resource doesn't exist. But not in this case so I simplified it.